app/Http/Controllers/SignupsController.php

```diff
@@ -121,7 +121,7 @@ public function show(Request $request, Signup $signup)
     {
         // Only allow an admin or the user who owns the signup to see the signup's unapproved posts.
         if ($request->query('include') === 'posts') {
-            $signup = Signup::withVisiblePosts()->first();
+            $signup = Signup::with('posts')->find($signup->id);
         }
```

Comment:
`$signup = $signup->visiblePosts();` erroring out makes sense because that is sending the relation, not the signup, to the transformer [here](https://github.com/DoSomething/rogue/blob/master/app/Http/Controllers/SignupsController.php#L127)?

Reply:
ah interesting! So `$signup = Signup::with('visiblePosts')->find($signup->id);` actually ends up calling `posts()` in `Signup.php` that returns all posts, regardless of status. Is this to be expected? cc @DoSomething/team-bleed

Reply:
I'm really fuzzy on this bug and I don't think I fully understand the `visiblePosts` thing. What is it that you are trying to accomplish and what is not happening?

Reply:
@sbsmith86 yes it's confusing! So for this endpoint, we'd like to grab a specific signup and include their posts (`/v3/signups/:signup_id?include=posts`). When using the `?include=posts` flag, the bug is that it returns the incorrect signup and was grabbing the first signup in the database.

I want to build a relation of `visiblePosts` for each signup here. The `visiblePosts` function will determine the following logic: 
- If you are an admin/staff, you can see all posts. 
- If you own the signup, you can see all posts. 
- If you aren't a admin/staff and you don't own the signup, you can only see `pending` posts. 

I'm trying to accomplish returning the `visiblePosts` based on the user but having trouble getting this to work - I'm not sure if I'm just doing something silly incorrectly based on the above? LMK if this is still unclear!

Reply:
That is super clear! Thank you for that and I'm sorry this is taking me a while to wrap my head around. 

* I don't think I see where the `visiblePosts` method is defined? is this a laravel convention?

* RE: the `posts()` method returning all posts regardless of status. It looks like we only apply a status filter to folks that are not staff members so if you are making the request as staff or an admin, I am guessing that is why the status is not being applied.

Reply:
Don't be - there has been a ton of conversation in this PR so it's hard to jump around - thanks for your patience! :) 

> I don't think I see where the `visiblePosts` method is defined? is this a laravel convention?
Sorry - again unclear! That is not in this PR (didn't push up the changes) - I was trying to build that out but was still returning all the posts (even pending for a non staff/admin/owner user). I realized that was because it actually then went through the `posts()` method in `Signup.php` after to return all of the posts. 

I think the best next step might be to try to get how it is working right now to work and dig into why it is breaking the v2 tests?

Reply:
yea i think move forward as you see best, and then we will take a look at the PR!

Reply:
@sbsmith86 the more I think about this, I think it's actually better to write this logic in it's own relation (not in `posts()` in `Signup.php`). 

The`v2` tests are breaking in the logic where it finds `posts_count` [here](https://github.com/DoSomething/rogue/blob/master/app/Repositories/Legacy/Two/PostRepository.php#L103) - since we aren't sure of the user's role in v2 endpoints, it returns `0` when we call `posts_count` with this logic. If I remove the new logic in this PR from the `posts()` method, it returns `1` when we call `posts_count`. 

For this reason, I think it's better to explicitly create a `visiblePosts` method and call that for this `v3` endpoint. 

However, this brings me back to the original problem! I can't seem to get the `visiblePosts` method working without going through the `posts()` method in `Signup.php` I'll push up what I have so it is easier to see.